### PR TITLE
Add build manual in README.md

### PR DESCRIPTION
**Pull request by genedna**
Add a build manual in README.md for newbie of Rust and RocksDB. Tested with [rocksdb 4.3.1](https://github.com/facebook/rocksdb/archive/rocksdb-4.3.1.tar.gz) release version and [gihub master version](https://github.com/facebook/rocksdb/commit/d02eb8d00b5beddd1a3e23d7def063e347bcd391). 

Signed-off-by: Meaglith Ma <genedna@gmail.com>

Close for #352 #348

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,6 +25,30 @@ TiKV uses Raft to support Geo-Replication. We have ported etcd's raft implementa
...
+
+wget https://github.com/facebook/rocksdb/archive/rocksdb-4.3.1.tar.gz
+tar -xzvf rocksdb-4.3.1.tar.gz
+
+# OR
+git clone https://github.com/facebook/rocksdb.git
+
+cd rocksdb
+make shared_lib
+cp cp librocksdb.so* /usr/lib64/
```
Comment (overvenus) [inline README.md]: cp cp? And what about 32bits OS?

Comment (genedna) [inline README.md]: @overvenus 

I haven't 32bit Linux OS for test. Maybe copy to  `lib` folder:

`cp librocksdb.so* /usr/lib/`

File: README.md [hunk 2]
```diff
@@ -31,10 +31,15 @@ curl -s https://static.rust-lang.org/rustup.sh | sh -s -- --channel=nightly
 
 # 2. Install RocksDB Library
 
+# 2.1 [optional step] Upgrade Your GCC To Version At Least 4.7 To Get C++11 Support
```
Comment (ngaut) [inline README.md]: I guess you may just add a reference to rocksdb's install guaid. https://github.com/facebook/rocksdb/blob/master/INSTALL.md

Comment (genedna) [inline README.md]: @ngaut Yeah, updated.

File: README.md [hunk 3]
```diff
@@ -51,7 +51,7 @@ git clone https://github.com/pingcap/tikv.git
 cd tikv
 make
 
-# 4. Have A Fun!
+# 4. Have a fun!
```
Comment (ngaut) [inline README.md]: s/a fun/fun

File: README.md [hunk 4]
```diff
@@ -26,20 +26,20 @@ TiKV uses Raft to support Geo-Replication. We have ported etcd's raft implementa
...
+# 1. Install rust nightly.
 curl -s https://static.rust-lang.org/rustup.sh | sh -s -- --channel=nightly
 
-# 2. Install RocksDB Library
-
-# 2.1 [optional step] Upgrade Your GCC To Version At Least 4.7 To Get C++11 Support
+# 2. Install RocksDB
+# Please view RocksDB's offical install document at https://github.com/facebook/rocksdb/blob/master/INSTALL.md .
 
+# 2.1 [optional step] Upgrade your GCC to version at least 4.7 to get C++11 support
```
Comment (ngaut) [inline README.md]: Since RocksDB already have the install manual, we should remove section 2.1. And merge section 2.2 to section 2.

**Comment by ngaut** [comment]
cc @paladintyrion PTAL

**Comment by genedna** [comment]
@ngaut The Travis CI which added an environment variable `COMPILER=clang++-3.6` build failed, it caused RocksDB build failed.

**Comment by ngaut** [comment]
Looks like clippy didn't catch up rust nightly. We will fix it later.

File: README.md (hunk 3 above)
Comment (ngaut) [inline README.md]: @genedna ping

File: README.md [hunk 5]
```diff
@@ -40,6 +40,32 @@ When node starts, the ids of node, store and region must be registered into pd a
...
+# Please view RocksDB's offical install document at https://github.com/facebook/rocksdb/blob/master/INSTALL.md . It's important to note, upgrade your GCC to version at least 4.7 to get C++11 support.
+wget https://github.com/facebook/rocksdb/archive/rocksdb-4.3.1.tar.gz
+tar -xzvf rocksdb-4.3.1.tar.gz
+
+# [optional step] clone the newest version of RocksDB
+git clone https://github.com/facebook/rocksdb.git
+
+cd rocksdb
+make shared_lib
+cp librocksdb.so* /usr/lib64/
```
Comment (siddontang) [inline README.md]: no need to copy rocksdb header file to /user/include ?

btw, I prefer /usr/local/lib, Mac has no `/usr/lib64` path. :-)

Comment (qiuyesuifeng) [inline README.md]: Can directly use `make install`?

Comment (siddontang) [inline README.md]: maybe yes, but may need `sudo`.

**Pull request closed without merging**